spec/services/api/v3/parse_query_params_service_spec.rb

```diff
@@ -319,5 +319,40 @@
         let(:expected) { { include_subprojects: false } }
       end
     end
+
+    context 'with timestamps' do
+      it_behaves_like 'transforms' do
+        let(:params) { { timestamps: "" } }
+        let(:expected) { { timestamps: [] } }
+      end
+
+      it_behaves_like 'transforms' do
+        let(:params) { { timestamps: "P-0Y" } }
+        let(:expected) { { timestamps: [Timestamp.parse("P-0Y")] } }
+      end
+
+      it_behaves_like 'transforms' do
+        let(:params) { { timestamps: "2022-10-29T23:01:23Z, P-0Y" } }
+        let(:expected) { { timestamps: [Timestamp.parse("2022-10-29T23:01:23Z"), Timestamp.parse("P-0Y")] } }
+      end
+
+      it_behaves_like 'transforms' do
+        let(:params) { { timestamps: "-1y, now" } }
+        let(:expected) { { timestamps: [Timestamp.new("P-1Y"), Timestamp.parse("P-0Y")] } }
```

Comment:
Ah, this is why you delegated the `Timestamp#hash` method :) 👍

Reply:
I added it for the [memoization where a timestamp is the hash key](https://github.com/opf/openproject/pull/12107/files#diff-c38c56127e4363f291ebe1c78dc9684b1d8d24d97cff6302f1ab5033e5cb2ff1R41)

I did not know that was necessary before stumbling over this. With it I can do
```
pry(main)> { Timestamp.now => 1}[Timestamp.now]
=> 1
```

Without it it would be
```
pry(main)> { Timestamp.now => 1}[Timestamp.now]
=> nil
```